### PR TITLE
Add the UI-only autorenewal toggle

### PR DESCRIPTION
**Pull request by southp**
#### Changes proposed in this Pull Request

*This PR is part of the attempt of adding a self-serving plan subscription autorenewal toggle. For more details, please refer to p2-p9jf6J-1GZ*

This PR adds a UI-only autorenewal toggle under the purchase management page, according to @nodeone 's design:
demo-renew-toggle

It is only enabled in the development environment through `autorenewal-toggle` feature flag. UI-only means that it does nothing for now. The actual functionalities will be wired in through the future PRs.

#### Testing instructions

Note that this is UI-only and is only enabled in the development environment.

* Log in as a user who has plan subscriptions and non-plan purchases
* Go to http://calypso.localhost:3000/me/purchases/
* Click on any plan subscriptions, see the toggle, and you should be able to toggle it as expected.
* Click on any non-plan purchases and there shouldn't be this toggle.

### CODE REVIEW

**Comment by michelleweber** [comment]
What is the copy that needs reviewing here? The toggle itself looks fine.

**Comment by southp** [comment]
@michelleweber Thanks for the fast check-in. You are right, it's pretty small and probably doesn't need much copy review:

* Auto-renew is ON / OFF
* You will be billed on yyyy/mm/dd
* Expires on yyyy/mm/dd

I guess I did that in the first place because I've developed the habit of labeling copy review whenever I touch any text, so it's kind of a reflex action 😅

File: client/me/purchases/manage-purchase/purchase-meta.jsx [hunk 1]
```diff
@@ -56,6 +58,17 @@ class PurchaseMeta extends Component {
 		purchaseId: false,
 	};
 
+	constructor( props ) {
+		super( props );
+
+		if ( config.isEnabled( 'autorenewal-toggle' ) ) {
+			// TODO: remove this once the proper state has been introduced.
+			this.state = {
```
Comment (spen) [inline client/me/purchases/manage-purchase/purchase-meta.jsx]: I'm a little hesitant with this approach as we can end up with or without `this.state` here, depending on the flag.
Of course, we only access `this.state` in logic that's guarded with the flag check, but it still makes me wonder if we can avoid potential future bugs.

Something like this comes to mind:

```js
state = {
  ...config.isEnabled( 'autorenewal-toggle' ) && {
    isAutorenewalEnabled: false,
  },
};
```

That way, `this.state` is always set to at least an empty object and future diffs can leave a much smaller change footprint:

```js
state = {
  someStateFromTheFuture: true,
  ...config.isEnabled( 'autorenewal-toggle' ) && {
    isAutorenewalEnabled: false,
  },
};
```

Comment (southp) [reply client/me/purchases/manage-purchase/purchase-meta.jsx]: Great advice! Added in 92fc8e4.

File: client/me/purchases/manage-purchase/purchase-meta.jsx [hunk 2]
```diff
@@ -287,12 +300,55 @@ class PurchaseMeta extends Component {
...
+			const subsRenewText = isAutorenewalEnabled
+				? translate( 'Auto-renew is ON' )
+				: translate( 'Auto-renew is OFF' );
+			const subsBillingText = isAutorenewalEnabled
+				? translate( 'You will be billed on %(renewDate)s', {
+						args: {
+							renewDate: purchase.renewMoment.format( 'LL' ),
+						},
+				  } )
+				: translate( 'Expires on %(expireDate)s', {
```
[the remark this answers is not in the document]
Comment (southp) [reply client/me/purchases/manage-purchase/purchase-meta.jsx]: Ah, nice catch. It wouldn't make any sense for expired subscriptions to turn on the auto-renewal, so I've excluded them in d987cf7. Now they will look just like what we have now:
螢幕快照 2019-05-17 下午3 17 05

**Comment by southp** [comment]
@niranjan-uma-shankar Good catch. It will be rolled out to G Suite in the future, but not now. I have exclude them in 3cf639b.

**Comment by southp** [comment]
@niranjan-uma-shankar @spen Would this UI tweak work for you?
demo-quarterly-divided

What I did in 0dcde1f 686f240 and were:
* Move the toggle down to the bottom.
* Divide the four into four equal, fixed width columns so they won't flick with the change of text.

**Comment by niranjan-uma-shankar** [comment]
Couple of observations:

1. When I tried this on a new test account, it works as described. However, when I tried it on my personal WPCOM account, clicking the toggle resulted in a blank screen: https://cloudup.com/cIDx-jAfdwo

The console gave the error:

```
purchase-meta.jsx:334 Uncaught TypeError: Cannot read property 'format' of null
    at PurchaseMeta.renderExpiration (purchase-meta.jsx:334)
    at PurchaseMeta.render (purchase-meta.jsx:389)
    at finishClassComponent (react-dom.development.js:14741)
    at updateClassComponent (react-dom.development.js:14696)
    at beginWork (react-dom.development.js:15644)
    at performUnitOfWork (react-dom.development.js:19312)
    at workLoop (react-dom.development.js:19352)
    at HTMLUnknownElement.callCallback (react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)
    at invokeGuardedCallback (react-dom.development.js:256)
```

I checked and noticed that the `renewMoment ` property takes a `null` value for my plan purchase. The only thing I can think of is different on my account is that I added a Premium upgrade to my account through the Store admin page. Do such upgrades give a null value for renewMoment?

If we do occasionally upgrade users via store admin, then this issue is worth a fix. Thoughts?

2. Not a strong opinion, but I think it would be better if the whole date breaks into a new line. For eg., I presently see this:

Screenshot 2019-05-20 at 10 35 41 AM

The date `9` is awkwardly sticking to the end of the line, and it could be easily missed.

I'm thinking it could be better if wrap the entire date in a <span> and add a `white-space: nowrap` style property to it:
`<span style="white-space:nowrap">9 April 2020</span>` makes it look like this:

Screenshot 2019-05-20 at 10 36 25 AM

**Comment by southp** [comment]
Great catch and wonderful suggestion on `nowrap`, @niranjan-uma-shankar !
I've addressed your feedbacks in 5a298c7 and 977440e, so it shouldn't crash for your test account anymore and the date string should stay on the same line. Please have a revisit to see if it works for you when you have some time 🙇

**Comment by southp** [comment]
@spen Ah! Thanks for the tip. I forgot we can use property initializer nowadays. Added in 84c3787  :)

**Pull request merged**